src/NuGetGallery.Core/Auditing/ReservedNamespaceAuditRecord.cs

```
@@ -0,0 +1,72 @@
+            : this(reservedNamespace, action, username: null, registrations: null)
+        { }
+
+        public ReservedNamespaceAuditRecord(ReservedNamespace reservedNamespace,
+            AuditedReservedNamespaceAction action,
+            string username,
+            IEnumerable<PackageRegistration> registrations)
+            : base(action)
+        {
+            if (reservedNamespace == null)
+            {
+                throw new ArgumentNullException(nameof(reservedNamespace));
+            }
+
+            Value = reservedNamespace.Value;
+            AffectedReservedNamespace = AuditedReservedNamespace.CreateFrom(reservedNamespace);
+            Action = action;
+
+            if (!string.IsNullOrWhiteSpace(username))
+            {
+                AffectedOwner = username;
+
+                var registrationAction = GetPackageRegistrationAction(action);
+                if (registrations != null && registrations.Any() && registrationAction.HasValue)
+                {
+                    AffectedRegistrations = registrations
+                        .Select(pr => new PackageRegistrationAuditRecord(pr, registrationAction.Value, username))
+                        .ToArray();
+                }
+            }
+        }
+
+        public override string GetPath()
+        {
+            return Value.ToLowerInvariant();
+        }
+
+        private static AuditedPackageRegistrationAction? GetPackageRegistrationAction(AuditedReservedNamespaceAction action)
+        {
+            switch (action)
+            {
+                case AuditedReservedNamespaceAction.AddOwner:
+                    return AuditedPackageRegistrationAction.MarkVerified;
+                case AuditedReservedNamespaceAction.RemoveOwner:
+                    return AuditedPackageRegistrationAction.MarkUnverified;
+                default:
+                    return null;
```

This default case shouldn't happen... I would have it throw instead of complicating the code to work with nullables.

---

Will throw and add tests.